src/modules/review/review.service.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  BadRequestException,
+  ConflictException,
+  ForbiddenException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { Review } from './entities/review.entity';
+import { CreateReviewDto } from './dto/create.review.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Booking } from '../bookings/entities/booking.entity';
+import { Repository } from 'typeorm';
+import { BookingStatus } from '../../shared/constants/booking-status.constant';
+import { Accommodation } from '../accommodations/entities/accommodations.entity';
+import { withBaseResponse } from '../../shared/utils/with-base-response.util';
+import { BaseResponse } from '../../shared/types/base-response.type';
+
+@Injectable()
+export class ReviewService {
+  constructor(
+    @InjectRepository(Booking)
+    private readonly bookingRepository: Repository<Booking>,
+    @InjectRepository(Review)
+    private readonly reviewRepository: Repository<Review>,
+    @InjectRepository(Accommodation)
+    private readonly accommodationRepository: Repository<Accommodation>,
+  ) {}
+
+  async createReview(
+    accountId: string,
+    createReviewDto: CreateReviewDto,
+  ): Promise<BaseResponse<Review>> {
+    const { accommodationId, bookingId, rating, content } = createReviewDto;
+
+    const booking = await this.bookingRepository.findOne({
+      where: { id: bookingId },
+    });
+    if (!booking) {
```

Comment:
Does ```booking``` have soft delete option? If so, you should check for that too

Reply:
I think deleting bookings hasn't been added yet. I don't know if it's a soft delete or not.

Reply:
Right now, booking does not have soft delete. But users can't review incomplete bookings: `booking.status !== BookingStatus.Completed`. And I suppose completed bookings should not be deleted. So no need for that.
This is only for checking invalid IDs sent by the frontend.